api.bs

```diff
@@ -1701,27 +1701,44 @@ To perform <dfn>common matching logic</dfn>, given
 
 <div algorithm>
 To <dfn>fairly allocate credit</dfn>, given a [=list=] of doubles |credit| and an integer |value|:
+1.  [=Assert=]: |credit| is not [=list/is empty|empty=].
 1.  Let |sumCredit| be the sum of |credit|'s [=list/items=].
-1.  Let |rawNormalizedCredit| be a new [=list=].
-1.  Let |normalizedCredit| be a new [=list=].
+1.  Let |roundedCredit| be a new [=list=].
 1.  [=list/iterate|For each=] |item| of |credit|:
-      1. Let |rawNormalized| be |value| * |item| / |sumCredit|.
-      1. [=list/Append=] |rawNormalized| to |rawNormalizedCredit|.
-      1. Let |normalized| be |rawNormalized| rounded towards positive Infinity and converted to an integer.
-      1. [=list/Append=] |normalized| to |normalizedCredit|.
-1.  Let |shuffledFractionalIndices| be [=list/get the indices|the indices=] of |credit|, ordered randomly.
-1.  [=list/remove|Remove=] all indices |i| from |shuffledFractionalIndices| where |rawNormalizedCredit|[|i|] has no fractional part.
-1.  [=list/iterate|For each=] |index| of |shuffledFractionalIndices|:
-      1. If the sum of |normalizedCredit|'s [=list/items=] is equal to |value|, [=iteration/break=].
-      1. Decrement |normalizedCredit|[|index|] by 1.
-1.  Return |normalizedCredit|.
+    1.  Let |normalizedCredit| be |value| * |item| / |sumCredit|.
+    1.  [=list/append|Append=] |normalizedCredit| to |roundedCredit|.
+1.  1. Let |idx1| be 0.
+1.  [=list/iterate|For each=] |n| of [=list/get the indices=] of |roundedCredit|, removing the first [=list/item=]:
+    1.  Let |idx2| be |n|.
+    1.  Let |frac1| be |roundedCredit|[|idx1|] - floor(|roundedCredit|[|idx1|]).
+    1.  Let |frac2| be |roundedCredit|[|idx2|] - floor(|roundedCredit|[|idx2|]).
+    1.  If |frac1| and |frac2| are both integers, [=iteration/continue=].
+    1.  If |frac1| + |frac2| is greater than 1, let |incr1| be 1-|frac1| and |incr2| be 1-|frac2|.
+    1.  Otherwise, let |incr1| be -|frac1| and |incr2| be -|frac2|.
```

Comment:
```suggestion
    1.  Let |frac1| be |roundedCredit|[|idx1|] − floor(|roundedCredit|[|idx1|]).
    1.  Let |frac2| be |roundedCredit|[|idx2|] − floor(|roundedCredit|[|idx2|]).
    1.  If |frac1| and |frac2| are both integers, [=iteration/continue=].
    1.  If |frac1| + |frac2| is greater than 1, let |incr1| be 1 − |frac1| and |incr2| be 1 − |frac2|.
    1.  Otherwise, let |incr1| be −|frac1| and |incr2| be −|frac2|.
```